### PR TITLE
refactor: update `blas/ext/base/sdsnansum` to follow current project conventions

### LINKED ISSUE #1152: Refactor and update existing extended base BLAS packages according to current project conventions
This should follow the same process as outlined in https://github.com/stdlib-js/stdlib/issues/788 but for `@stdlib/blas/ext/base/*`.

Packages needing updating:

- [x] `@stdlib/blas/ext/base/dapx`
- [x] `@stdlib/blas/ext/base/dapxsum`
- [x] `@stdlib/blas/ext/base/dapxsumkbn`
- [x] `@stdlib/blas/ext/base/dapxsumkbn2`
- [x] `@stdlib/blas/ext/base/dapxsumors`
- [x] `@stdlib/blas/ext/base/dapxsumpw`
- [x] `@stdlib/blas/ext/base/dasumpw`
- [x] `@stdlib/blas/ext/base/dcusum`
- [x] `@stdlib/blas/ext/base/dcusumkbn`
- [x] `@stdlib/blas/ext/base/dcusumkbn2`
- [x] `@stdlib/blas/ext/base/dcusumors`
- [x] `@stdlib/blas/ext/base/dcusumpw`
- [x] `@stdlib/blas/ext/base/dfill`
- [x] `@stdlib/blas/ext/base/dnanasum`
- [x] `@stdlib/blas/ext/base/dnanasumors`
- [x] `@stdlib/blas/ext/base/dnannsum`
- [x] `@stdlib/blas/ext/base/dnannsumkbn`
- [x] `@stdlib/blas/ext/base/dnannsumkbn2`
- [x] `@stdlib/blas/ext/base/dnannsumors`
- [x] `@stdlib/blas/ext/base/dnannsumpw`
- [x] `@stdlib/blas/ext/base/dnansum`
- [x] `@stdlib/blas/ext/base/dnansumkbn`
- [x] `@stdlib/blas/ext/base/dnansumkbn2`
- [x] `@stdlib/blas/ext/base/dnansumors`
- [x] `@stdlib/blas/ext/base/dnansumpw`
- [x] `@stdlib/blas/ext/base/drev`
- [x] `@stdlib/blas/ext/base/dsapxsum`
- [x] `@stdlib/blas/ext/base/dsapxsumpw`
- [x] `@stdlib/blas/ext/base/dsnannsumors`
- [x] `@stdlib/blas/ext/base/dsnansum`
- [x] `@stdlib/blas/ext/base/dsnansumors`
- [x] `@stdlib/blas/ext/base/dsnansumpw`
- [x] `@stdlib/blas/ext/base/dsort2hp`
- [x] `@stdlib/blas/ext/base/dsort2ins`
- [x] `@stdlib/blas/ext/base/dsort2sh`
- [x] `@stdlib/blas/ext/base/dsorthp`
- [x] `@stdlib/blas/ext/base/dsortins`
- [x] `@stdlib/blas/ext/base/dsortsh`
- [x] `@stdlib/blas/ext/base/dssum`
- [x] `@stdlib/blas/ext/base/dssumors`
- [x] `@stdlib/blas/ext/base/dssumpw`
- [x] `@stdlib/blas/ext/base/dsum`
- [x] `@stdlib/blas/ext/base/dsumkbn`
- [x] `@stdlib/blas/ext/base/dsumkbn2`
- [x] `@stdlib/blas/ext/base/dsumors`
- [x] `@stdlib/blas/ext/base/dsumpw`
- [x] `@stdlib/blas/ext/base/sapx`
- [x] `@stdlib/blas/ext/base/sapxsum`
- [x] `@stdlib/blas/ext/base/sapxsumkbn`
- [x] `@stdlib/blas/ext/base/sapxsumkbn2`
- [x] `@stdlib/blas/ext/base/sapxsumors`
- [x] `@stdlib/blas/ext/base/sapxsumpw`
- [x] `@stdlib/blas/ext/base/sasumpw`
- [x] `@stdlib/blas/ext/base/scusum`
- [x] `@stdlib/blas/ext/base/scusumkbn`
- [x] `@stdlib/blas/ext/base/scusumkbn2`
- [x] `@stdlib/blas/ext/base/scusumors`
- [x] `@stdlib/blas/ext/base/scusumpw`
- [x] `@stdlib/blas/ext/base/sdsapxsum`
- [x] `@stdlib/blas/ext/base/sdsapxsumpw`
- [x] `@stdlib/blas/ext/base/sdsnansum`
- [x] `@stdlib/blas/ext/base/sdsnansumpw`
- [x] `@stdlib/blas/ext/base/sdssum`
- [x] `@stdlib/blas/ext/base/sdssumpw`
- [x] `@stdlib/blas/ext/base/sfill`
- [x] `@stdlib/blas/ext/base/snansum`
- [x] `@stdlib/blas/ext/base/snansumkbn`
- [x] `@stdlib/blas/ext/base/snansumkbn2`
- [x] `@stdlib/blas/ext/base/snansumors`
- [x] `@stdlib/blas/ext/base/snansumpw`
- [x] `@stdlib/blas/ext/base/srev`
- [x] `@stdlib/blas/ext/base/ssort2hp`
- [x] `@stdlib/blas/ext/base/ssort2ins`
- [x] `@stdlib/blas/ext/base/ssort2sh`
- [x] `@stdlib/blas/ext/base/ssorthp`
- [x] `@stdlib/blas/ext/base/ssortins`
- [x] `@stdlib/blas/ext/base/ssortsh`
- [x] `@stdlib/blas/ext/base/ssum`
- [x] `@stdlib/blas/ext/base/ssumkbn`
- [x] `@stdlib/blas/ext/base/ssumkbn2`
- [x] `@stdlib/blas/ext/base/ssumors`
- [x] `@stdlib/blas/ext/base/ssumpw`

### LINKED ISSUE #1523: [RFC]: refactor `blas/ext/base/sdsnansum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/sdsnansum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/sdsnansum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### LINKED ISSUE #1533: [RFC]: refactor `blas/ext/base/srev` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/srev`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/srev) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by alextes90**
Resolves #1523 .

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors blas/ext/base/sdsnansum to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1523 
-   fixes #1152 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/README.md [hunk 1]
```diff
@@ -139,22 +132,14 @@ var v = sdsnansum.ndarray( N, x, 2, 1 );
...
-    if ( randu() < 0.2 ) {
-        x[ i ] = NaN;
-    } else {
-        x[ i ] = round( randu()*100.0 );
-    }
-}
+
+var rand = discreteUniform( 0, 100 );
+var i = filledarrayBy( len, 'float32', rand );
+var x = i.map(el => discreteUniform( 0.0, 1.0 ) < 0.2 ? el : NaN)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/README.md]: This isn't necessary. Provide a single function to `filledarrayBy` which handles both `rand` and `NaN` value generation. Meaning, we can create a filled array in a single pass.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js [hunk 2]
```diff
@@ -39,37 +44,28 @@ var sdsnansum = require( './../lib/sdsnansum.js' );
...
-		}
-		b.toc();
-		if ( isnan( v ) ) {
-			b.fail( 'should not return NaN' );
-		}
-		b.pass( 'benchmark finished' );
-		b.end();
-	}
+  var i = filledarrayBy( len, 'float32', rand );
+  var x = i.map(el => uniform( 0.0, 1.0 ) < 0.2 ? el : NaN)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js]: Same comment as above. Also, we use TAB indentation in JS files here and everywhere else in this PR.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js [hunk 3]
```diff
@@ -81,20 +77,20 @@ function createBenchmark( len ) {
...
-
-	min = 1; // 10^min
-	max = 6; // 10^max
-
-	for ( i = min; i <= max; i++ ) {
-		len = pow( 10, i );
-		f = createBenchmark( len );
-		bench( pkg+':len='+len, f );
-	}
+  var len;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js]: Don't change indentation.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.native.js [hunk 4]
```diff
@@ -48,17 +49,8 @@ var opts = {
...
-
-	x = new Float32Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
-		}
-	}
+	var i = filledarrayBy( len, 'float32', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.native.js]: Same comment as above.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/README.md (hunk 1 above)
Comment (alextes90) [reply lib/node_modules/@stdlib/blas/ext/base/sdsnansum/README.md]: Have changed

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js (hunk 2 above)
Comment (alextes90) [reply lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js]: Have changed

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js (hunk 3 above)
Comment (alextes90) [reply lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.js]: Fixed

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.native.js (hunk 4 above)
Comment (alextes90) [reply lib/node_modules/@stdlib/blas/ext/base/sdsnansum/benchmark/benchmark.native.js]: Have changed

**Comment by alextes90** [comment]
Fix all errors

**Comment by alextes90** [comment]
Forth attempt)

**Comment by alextes90** [comment]
I make the change but I am not sure where the problem come from and why the system failed to build addon. 
Can you help with it?

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/manifest.json [hunk 5]
```diff
@@ -1,42 +1,79 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/blas/ext/base/sdsnansumpw"]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/manifest.json]:
```suggestion
        "@stdlib/blas/ext/base/sdsnansumpw"
      ]
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/manifest.json [hunk 6]
```diff
@@ -1,42 +1,79 @@
...
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/blas/ext/base/sdsnansumpw",
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-strided-float32array"
+      ]
+    },    {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/manifest.json]:
```suggestion
    },
    {
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt [hunk 7]
```diff
@@ -1,9 +1,11 @@
 
 {{alias}}( N, x, stride )
-    Computes the sum of single-precision floating-point strided array elements,
-    ignoring `NaN` values and using extended accumulation.
+    Computes the sum of single-precision
+    floating-point strided array elements,
+    ignore `NaN` values and using extended accumulation.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt]: This is not properly wrapped at 80 chars. See https://github.com/stdlib-js/stdlib/blob/develop/docs/repl_text.md#wrapping

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/types/index.d.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/types/index.d.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

This file was not modified. No need to change the copyright year.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/src/addon.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, stdlib_strided_sdsnansum( N, X, stride ), &v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/src/addon.c]:
```suggestion
	napi_status status = napi_create_double( env, (double)stdlib_strided_sdsnansum( N, X, stride ), &v );
```

`sdsnansum` returns a `float`, IIRC, so we make explicit the cast.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/test/test.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/test/test.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt [hunk 11]
```diff
@@ -86,8 +87,7 @@
 
     // Using offset parameter:
     > var x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0, NaN, NaN ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt]:
```suggestion
    > x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0, NaN, NaN ] );
```

**Comment by kgryte** [comment]
@alextes90 Looks like the addon is successfully building now. Seems you were just missing dependencies in the `manifest.json` file.

File: lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt (hunk 7 above)
Comment (alextes90) [reply lib/node_modules/@stdlib/blas/ext/base/sdsnansum/docs/repl.txt]: Have fixed

**Review by Planeshifter — APPROVED** [review]
Looks all good now. Will merge after CI checks have passed; I only restored the original copyright years in a bunch of files (we don't change them for existing files) and updated how the random numbers are generated to mirror how we have done it in a bunch of the other `blas/ext/base/*nan*` packages. Thanks @alextes90!

**Pull request merged**